src/Orleans.Runtime/Lifecycle/SiloLifecycleSubject.cs

```diff
@@ -12,38 +12,38 @@ namespace Orleans.Runtime
     /// <summary>
     /// Decorator over lifecycle subject for silo.  Adds some logging and monitoring
     /// </summary>
-    public class SiloLifecycleSubject : ISiloLifecycleSubject
+    public class SiloLifecycleSubject : LifecycleSubject, ISiloLifecycleSubject
     {
-        private readonly ILifecycleSubject subject;
         private readonly ILogger<SiloLifecycleSubject> logger;
         private readonly List<MonitoredObserver> observers;
 
-        public SiloLifecycleSubject(ILifecycleSubject subject, ILogger<SiloLifecycleSubject> logger)
+        public SiloLifecycleSubject(ILogger<SiloLifecycleSubject> logger)
+            :base(logger)
         {
-            this.subject = subject;
             this.logger = logger;
             this.observers = new List<MonitoredObserver>();
         }
 
-        public Task OnStart(CancellationToken ct)
+        private void PerfMeasure(int? stage, long timeLapsed)
         {
-            foreach(IGrouping<int,MonitoredObserver> stage in this.observers.GroupBy(o => o.Stage).OrderBy(s => s.Key))
-            {
-                this.logger?.Info(ErrorCode.LifecycleStagesReport, $"Stage {stage.Key}: {string.Join(", ", stage.Select(o => o.Name))}", stage.Key);
-            }
-            return this.subject.OnStart(ct);
+             this.logger?.Info(ErrorCode.SiloStartPerfMeasure, $"Starting lifecycle stage {stage} took {timeLapsed} Milliseconds");
+        }
+
+        public override Task OnStart(CancellationToken ct)
+        {
+            return this.OnStart(ct, this.PerfMeasure);
         }
 
-        public Task OnStop(CancellationToken ct)
+        public override  Task OnStop(CancellationToken ct)
         {
-            return this.subject.OnStop(ct);
+            return this.OnStop(ct, this.PerfMeasure);
         }
 
-        public IDisposable Subscribe(string observerName, int stage, ILifecycleObserver observer)
+        public override IDisposable Subscribe(string observerName, int stage, ILifecycleObserver observer)
         {
             var monitoredObserver = new MonitoredObserver(observerName, stage, observer, this.logger);
             this.observers.Add(monitoredObserver);
-            return this.subject.Subscribe(observerName, stage, monitoredObserver);
+            return this.Subscribe(observerName, stage, monitoredObserver);
```

Comment:
isn't this recursive?
base.Subscribe?

Reply:
sorry didn't mean to submit this change... but submit to avoid my cp=omputer got moved and lost all the datas. wikll fix